CI failure in codegen - Helm 3.17.1 allows overwrite table with non table for null fields in values.yaml

<!-- If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack). -->

Checklist:

* [ ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

PRs failing on `make codegen-local ` at the step of `Check changes to generated code`.
The failure happens upon running `helm dependency update ./chart` in `manifests/ha/base/redis-ha/generate.sh`, because the `argo-cd/manifests/ha/base/redis-ha/chart/values.yaml` contains `containerSecurityContext: null`  and this SUDDENLY seems to override with null  the securityContext  in the `manifests/ha/base/redis-ha/chart/upstream.yaml` , which complains both at running the command and at the diff that is suddenly generated.

There are 2 issues that cause this:

1. `containerSecurityContext: null` is not valid for Redis and so `helm template argocd ./chart --namespace argocd --values ./chart/values.yaml --no-hooks` gives a warning:
`coalesce.go:286: warning: cannot overwrite table with non table for redis-ha.redis-ha.haproxy.containerSecurityContext (map[allowPrivilegeEscalation:false capabilities:map[drop:[ALL]] runAsNonRoot:true seccompProfile:map[type:RuntimeDefault]])`
2. In Helm < 3.17.1 , the `helm template` command DOES NOT override the `upstream.yaml` file, so the warning actually has no impact. In Helm 3.17.1, the warning is still there, but the `helm template` command DOES override the file, in other words allowing the invalid value on the manifests, and this causes a diff and a failure in the CI step.

The solution seems to be a removal of the `containerSecurityContext: null` from `values.yaml` for redis-ha.

**To Reproduce**

<!-- A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue. -->

**Expected behavior**

<!-- A clear and concise description of what you expected to happen. -->

**Screenshots**

<!-- If applicable, add screenshots to help explain your problem. -->

**Version**

```shell
Paste the output from `argocd version` here.
```

**Logs**

```
Paste any relevant application logs here.
```

--- a/manifests/ha/base/redis-ha/chart/values.yaml
+++ b/manifests/ha/base/redis-ha/chart/values.yaml
@@ -15,7 +15,6 @@ redis-ha:
       enabled: false
     image:
       tag: 2.6.17-alpine
-    containerSecurityContext: null
     timeout:
       server: 6m
       client: 6m
@@ -24,7 +23,6 @@ redis-ha:
       enabled: true
   image:
     tag: 7.0.15-alpine
-  containerSecurityContext: null
   sentinel:
     bind: '0.0.0.0'
     lifecycle:

